src/app/container/version-modal/version-modal.component.ts

```diff
@@ -277,20 +277,19 @@ export class VersionModalComponent implements OnInit, AfterViewChecked {
     return this.dateService.getDateTimeMessage(timestamp);
   }
 
+  // Validation ends here
   // Checks if the currently edited test parameter file already exists
-  hasDuplicateTestJson(type) {
+  // TODO: This code is repeated in add-tag.component.ts for tools, move it somewhere common
+  // TODO: This is also executed a bajillion times
+  hasDuplicateTestJson(type: ToolDescriptor.TypeEnum) {
     if (type === this.DescriptorType.CWL) {
-      if (this.unsavedCWLTestParameterFilePaths.includes(this.unsavedTestCWLFile)) {
-        return true;
-      } else {
-        return false;
-      }
+      return this.unsavedWDLTestParameterFilePaths.includes(this.unsavedTestCWLFile) ||
```

Comment:
same deal, except in this case, might be possible to share a utility function
looks like a copy and paste

Reply:
ya, it's in the TODO already.  didn't want to keep stretching this misc PR any further

Reply:
sounds good, then create an issue to remember and then move on